### PR TITLE
VIH-9612 Better error handling for get audio recorders

### PR DESCRIPTION
**Pull request by will-craig**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-9612

### Change description ###
The main change here is to individually wrap each request to the wowza server in its own try catch (rather than the entire method) to allow it, if it catches an unexpected error, to carry on. In the scenario that the client cant find the end point, rather than return a HttpResponseMessage object, with 404 (which would be handled normally), it was continuing to poll for 100 seconds before timing out, and the video-api would then throw an exception overiding the OK response from other node. Have also added a 10 second cancellation token to the request, so that the task doesnt continue to run in the background, if the other has returned ok. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by marcogagliardi — APPROVED** [review]
👌🏽

File: VideoApi/VideoApi.Services/AudioPlatformService.cs [hunk 1]
```diff
@@ -79,28 +79,30 @@ public async Task<WowzaMonitorStreamResponse> GetAudioStreamMonitoringInfoAsync(
...
-                _logger.LogInformation("Got Wowza stream recorder for application: {recorder}", recorder);
-                return JsonConvert.DeserializeObject<WowzaGetStreamRecorderResponse>(await successfulResponse.Content.ReadAsStringAsync());
-            }
-            catch (AudioPlatformException ex)
-            {
-                var errorMessageTemplate = "Failed to get the Wowza stream recorder for: {recorder}, StatusCode: {ex.StatusCode}, Error: {ex.Message}";                
-                LogError(ex, errorMessageTemplate, recorder, ex.StatusCode, ex.Message);
-                return null;
             }
+            var errorMessage = await responses.First().Content.ReadAsStringAsync();
```
Comment (shaed-parkar) [inline VideoApi/VideoApi.Services/AudioPlatformService.cs:102]: I think an aggregate exception would be better so then you can display all the exceptions, incase the error is different

**Review by shaed-parkar — COMMENTED** [review]
you're handling the error better but losing them by only using the first.

**Review by marcogagliardi — APPROVED** [review]
👌🏽

**Pull request merged**